pandas/_libs/window/aggregations.pyx

```diff
@@ -198,11 +199,10 @@ def roll_sum_variable(ndarray[float64_t] values, ndarray[int64_t] start,
             s = start[i]
             e = end[i]
 
-            if i == 0:
+            if i == 0 or not is_monotonic_bounds:
```

Comment:
could we do a simple calculcate in start/end to determine if is_monotonic is true here? it likely is quite performant.